Subject: lib: on OpenVMS, include the generated *.gl.h headers as *_gl.h

VSI C cannot #include a file whose name has two dots (on ODS-5 the file is
dynarray^.gl.h).  tools/prepare.sh renames the generated headers to
*_gl.h; include them by that name on VMS.  (As in vms-sed.)

--- a/lib/dynarray.h	2026-10-05 02:51:41.907704455 +0100
+++ b/lib/dynarray.h	2026-10-05 02:51:41.948229635 +0100
@@ -265,14 +265,22 @@
 
 /* Define auxiliary structs and declare auxiliary functions, common to all
    instantiations of dynarray.  */
-# include <malloc/dynarray.gl.h>
+# ifdef __VMS
+#  include <malloc/dynarray_gl.h>
+# else
+#  include <malloc/dynarray.gl.h>
+# endif
 
 /* Define the instantiation, specified through
      DYNARRAY_STRUCT
      DYNARRAY_ELEMENT
      DYNARRAY_PREFIX
    etc.  */
-# include <malloc/dynarray-skeleton.gl.h>
+# ifdef __VMS
+#  include <malloc/dynarray-skeleton_gl.h>
+# else
+#  include <malloc/dynarray-skeleton.gl.h>
+# endif
 
 #else
 
